modules/gdnative/nativearvr/register_types.cpp

```diff
@@ -0,0 +1,427 @@
+/*************************************************************************/
+/*  register_types.cpp                                                   */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2017 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2017 Godot Engine contributors (cf. AUTHORS.md)    */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "register_types.h"
+
+#include "arvr_interface_gdnative.h"
+#include "core/os/os.h"
+
+typedef void *void_ptr;
+typedef real_t *real_t_ptr;
```

Comment:
What's with those typedefs?

I think `void *` is easier to read than `void_ptr`, I don't see value in this.

Reply:
Weirdly enough, it was the only way I could get lines like this to work:
void_ptr *return_ptr = (void_ptr *)r_return;

Without the typedef my compiler wouldn't let me use a void **, kept complaining about casting voids.

Reply:
It should work without the typedef, I wrote a suggestion for the function pointers in another comment.

`void **return_ptr = (void **)r_return;` has to work though :P

Reply:
The function pointers I just copied for consistency, using void * works fine there.

I'll try the line of code you mentioned again tonight, I'm pretty sure thats what I started with but after trying several combinations with my compiler spitting the dummy at me... 
I agree with you, that notation should just work, maybe it's just something weird Apple dictates in their compilers?

Anyway I'll try again and if it doesn't work I'll post the compilers complaints, and if it does work, I fully agree with you, these typedefs are ugly :)

Reply:
beats me why it wouldn't work last night, probably just too late and my brain not functioning...